### PR TITLE
edit last

### PR DESCRIPTION
**Pull request by ge0rge000**

### CODE REVIEW

File: src/main/java/helloworldapp/HelloWorldWorkflowImpl.java [hunk 1]
```diff
@@ -21,6 +21,6 @@
     public String getGreeting(String name) {
       
 
-        return activities.composeActivity("Hello", name);
+        return currency.receiveApi(name);
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/helloworldapp/HelloWorldWorkflowImpl.java:24]: Remove unused activity declarations.

File: src/main/java/url/ApiInterface.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.Context;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import io.vertx.ext.web.codec.BodyCodec;
+import io.vertx.core.Vertx;
+public class ApiInterface extends AbstractVerticle {
+
+
+    public static void main(String[ ] args){
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/url/ApiInterface.java:23]: what does this file do?

File: src/main/java/helloworldapp/HelloWorldWorkflowImpl.java (hunk 1 above)
Comment (ge0rge000) [reply src/main/java/helloworldapp/HelloWorldWorkflowImpl.java:24]: ok i will remove

File: src/main/java/helloworldapp/InitiateHelloWorld.java [hunk 3]
```diff
@@ -23,7 +23,9 @@ public void start() throws Exception {
         HelloWorldWorkflow workflow = client.newWorkflowStub(HelloWorldWorkflow.class, options);
         // Synchronously execute the Workflow and wait for the response.
         CompletableFuture<String> greeting = WorkflowClient.execute(workflow::getGreeting, "World");
-        System.out.println(greeting.get());
+       System.out.println(greeting.get());
+//        String greeting = workflow.getGreeting("World");
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/helloworldapp/InitiateHelloWorld.java:27]: Remove commented code.

File: src/main/java/url/ApiInterface.java (hunk 2 above)
Comment (ge0rge000) [reply src/main/java/url/ApiInterface.java:23]: to test before enter in temporal files

File: src/main/java/helloworldapp/InitiateHelloWorld.java (hunk 3 above)
Comment (ge0rge000) [reply src/main/java/helloworldapp/InitiateHelloWorld.java:27]: ok

File: src/main/java/url/ApiInterface.java (hunk 2 above)
Comment (Kirollos-Hanna-Yodawy) [reply src/main/java/url/ApiInterface.java:23]: remove it if you're no longer using it.

File: src/main/java/helloworldapp/VertxVerticleMain.java [hunk 4]
```diff
@@ -6,7 +6,7 @@
 import io.vertx.core.Vertx;
 import helloworldapp.InitiateHelloWorld;
 import helloworldapp.HelloWorldWorker;
-public class VertxVerticleMain {
+public class VertxVerticleMain  {
 
     public static void main(String[] args) {
         Vertx vertx = Vertx.vertx();
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/helloworldapp/VertxVerticleMain.java:12]: create a mainVerticle file that deploys helloWorldWorkerVerticle and InitiateHelloWorldVerticle. Deploy mainVerticle from this file.

Also, rename the verticle files to follow convention (add Verticle to the end of their name)

File: src/main/java/helloworldapp/VertxVerticleMain.java [hunk 5]
```diff
@@ -6,7 +6,7 @@
 import io.vertx.core.Vertx;
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/helloworldapp/VertxVerticleMain.java:6]: Remove unused imports

Comment (ge0rge000) [reply src/main/java/helloworldapp/VertxVerticleMain.java:6]: ok

File: src/main/java/helloworldapp/VertxVerticleMain.java (hunk 4 above)
Comment (ge0rge000) [reply src/main/java/helloworldapp/VertxVerticleMain.java:12]: ok

File: src/main/java/helloworldapp/Apimpl.java [hunk 6]
```diff
@@ -1,35 +1,46 @@
...
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import io.vertx.ext.web.codec.BodyCodec;
 public class Apimpl implements Api {
+    private final ActivityCompletionClient completionClient;
+
+    Apimpl(ActivityCompletionClient completionClient) {
+        this.completionClient = completionClient;
+    }    @Override
+
+    public String receiveApi(String currency,Integer price) {
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/helloworldapp/Apimpl.java:24]: where is the currency parameter being used?

File: src/main/java/helloworldapp/HelloWorldWorker.java [hunk 7]
```diff
@@ -23,7 +23,7 @@ public void start() throws Exception {
         worker.registerWorkflowImplementationTypes(HelloWorldWorkflowImpl.class);
         // Activities are stateless and thread safe, so a shared instance is used.
         ActivityCompletionClient completionClient = client.newActivityCompletionClient();
-        worker.registerActivitiesImplementations(new FormatImpl(),new Apimpl(),new GreetingActivitiesImpl(completionClient));
+        worker.registerActivitiesImplementations(new FormatImpl(),new Apimpl(completionClient),new GreetingActivitiesImpl(completionClient));
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/helloworldapp/HelloWorldWorker.java:26]: add the new activity to your worker

File: src/main/java/helloworldapp/PostApi.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+
+package helloworldapp;
+
+        import io.temporal.activity.ActivityInterface;
+        import io.temporal.activity.ActivityMethod;
+
+@ActivityInterface
+public interface PostApi {
+    String convert = "";
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/helloworldapp/PostApi.java:9]: what is this variable?

File: src/main/java/helloworldapp/PostApimpl.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import io.vertx.ext.web.codec.BodyCodec;
+public class PostApimpl implements PostApi  {
+   static String convert;
+    private final ActivityCompletionClient completionClient;
+
+    PostApimpl(ActivityCompletionClient completionClient) {
+        this.completionClient = completionClient;
+    }
+        @Override
+    public String postApi(){
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/helloworldapp/PostApimpl.java:29]: Review this with Eslam and check if that is what the requirement is.

File: src/main/java/helloworldapp/post_Vertx.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+import io.vertx.ext.web.Router;
+
+public class post_Vertx  {
+    public static String type="USD";
+    public static void main(String[] args) {
```
Comment (Kirollos-Hanna-Yodawy) [inline src/main/java/helloworldapp/post_Vertx.java:12]: Review this with Eslam and check if that is what the requirement is.

**Pull request merged**